Error while checking build/O2/fullCI for 40e7d6bafab59bf153f302574d7b9d2e9e069c9f at 2024-03-22 17:28:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:224:17: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:228:17: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:233:24: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:237:24: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:284:22: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:286:22: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:293:32: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:295:32: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:300:33: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:302:33: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:307:33: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:309:33: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:358:32: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
/sw/SOURCES/O2/12871-slc8_x86-64/0/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx:360:32: error: cannot convert 'TH1F*' to 'TH1D*' in assignment
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12871/40e7d6bafab59bf153f302574d7b9d2e9e069c9f/build_O2_fullCI/pretty.html).

--- a/Detectors/GlobalTracking/include/GlobalTracking/MatchITSTPCQC.h
+++ b/Detectors/GlobalTracking/include/GlobalTracking/MatchITSTPCQC.h
@@ -136,6 +136,7 @@ class MatchITSTPCQC
   void publishHistograms(const std::shared_ptr<T>& publisher)
   {
     for (int i = 0; i < matchType::SIZE; ++i) {
+      // Pt
       publisher->startPublishing(mPtNum[i]);
       publisher->startPublishing(mPtDen[i]);
       publisher->startPublishing(mFractionITSTPCmatch[i]);
@@ -144,13 +145,38 @@ class MatchITSTPCQC
       publisher->startPublishing(mPtDen_noEta0[i]);
       publisher->startPublishing(mFractionITSTPCmatch_noEta0[i]);
 
-      publisher->startPublishing(mPtPhysPrimNum[i]);
-      publisher->startPublishing(mPtPhysPrimDen[i]);
-      publisher->startPublishing(mFractionITSTPCmatchPhysPrim[i]);
-
+      // Phi
       publisher->startPublishing(mPhiNum[i]);
       publisher->startPublishing(mPhiDen[i]);
       publisher->startPublishing(mFractionITSTPCmatchPhi[i]);
+
+      publisher->startPublishing(mPhiVsPtNum[i]);
+      publisher->startPublishing(mPhiVsPtDen[i]);
+      publisher->startPublishing(mFractionITSTPCmatchPhiVsPt[i]);
+
+      // Eta
+      publisher->startPublishing(mEtaNum[i]);
+      publisher->startPublishing(mEtaDen[i]);
+      publisher->startPublishing(mFractionITSTPCmatchEta[i]);
+
+      publisher->startPublishing(mEtaVsPtNum[i]);
+      publisher->startPublishing(mEtaVsPtDen[i]);
+      publisher->startPublishing(mFractionITSTPCmatchEtaVsPt[i]);
+
+      // Clusters
+      publisher->startPublishing(mClsVsPtNum[i]);
+      publisher->startPublishing(mClsVsPtDen[i]);
+      publisher->startPublishing(mFractionITSTPCmatchClsVsPt[i]);
+
+      // Chi2
+      publisher->startPublishing(mChi2VsPtNum[i]);
+      publisher->startPublishing(mChi2VsPtDen[i]);
+      publisher->startPublishing(mFractionITSTPCmatchChi2VsPt[i]);
+
+      // 1/pt
+      publisher->startPublishing(m1OverPtNum[i]);
+      publisher->startPublishing(m1OverPtDen[i]);
+      publisher->startPublishing(mFractionITSTPCmatch1OverPt[i]);
 
       if (mUseTrkPID) { // Vs Tracking PID hypothesis
         for (int j = 0; j < o2::track::PID::NIDs; ++j) {
@@ -158,10 +184,12 @@ class MatchITSTPCQC
           publisher->startPublishing(mPtNumVsTrkPID[i][j]);
           publisher->startPublishing(mPtDenVsTrkPID[i][j]);
           publisher->startPublishing(mFractionITSTPCmatchPtVsTrkPID[i][j]);
+
           // Phi
           publisher->startPublishing(mPhiNumVsTrkPID[i][j]);
           publisher->startPublishing(mPhiDenVsTrkPID[i][j]);
           publisher->startPublishing(mFractionITSTPCmatchPhiVsTrkPID[i][j]);
+
           // Eta
           publisher->startPublishing(mEtaNumVsTrkPID[i][j]);
           publisher->startPublishing(mEtaDenVsTrkPID[i][j]);
@@ -169,41 +197,20 @@ class MatchITSTPCQC
         }
       }
 
-      publisher->startPublishing(mPhiPhysPrimNum[i]);
-      publisher->startPublishing(mPhiPhysPrimDen[i]);
-      publisher->startPublishing(mFractionITSTPCmatchPhiPhysPrim[i]);
-
-      publisher->startPublishing(mPhiVsPtNum[i]);
-      publisher->startPublishing(mPhiVsPtDen[i]);
-      publisher->startPublishing(mFractionITSTPCmatchPhiVsPt[i]);
-
-      publisher->startPublishing(mEtaNum[i]);
-      publisher->startPublishing(mEtaDen[i]);
-      publisher->startPublishing(mFractionITSTPCmatchEta[i]);
-
-      publisher->startPublishing(mEtaPhysPrimNum[i]);
-      publisher->startPublishing(mEtaPhysPrimDen[i]);
-      publisher->startPublishing(mFractionITSTPCmatchEtaPhysPrim[i]);
-
-      publisher->startPublishing(mEtaVsPtNum[i]);
-      publisher->startPublishing(mEtaVsPtDen[i]);
-      publisher->startPublishing(mFractionITSTPCmatchEtaVsPt[i]);
-
-      publisher->startPublishing(mClsVsPtNum[i]);
-      publisher->startPublishing(mClsVsPtDen[i]);
-      publisher->startPublishing(mFractionITSTPCmatchClsVsPt[i]);
-
-      publisher->startPublishing(mChi2VsPtNum[i]);
-      publisher->startPublishing(mChi2VsPtDen[i]);
-      publisher->startPublishing(mFractionITSTPCmatchChi2VsPt[i]);
-
-      publisher->startPublishing(m1OverPtNum[i]);
-      publisher->startPublishing(m1OverPtDen[i]);
-      publisher->startPublishing(mFractionITSTPCmatch1OverPt[i]);
-
-      publisher->startPublishing(m1OverPtPhysPrimNum[i]);
-      publisher->startPublishing(m1OverPtPhysPrimDen[i]);
-      publisher->startPublishing(mFractionITSTPCmatchPhysPrim1OverPt[i]);
+      if (mUseMC) {
+        publisher->startPublishing(mPhiPhysPrimNum[i]);
+        publisher->startPublishing(mPhiPhysPrimDen[i]);
+        publisher->startPublishing(mFractionITSTPCmatchPhiPhysPrim[i]);
+        publisher->startPublishing(mPtPhysPrimNum[i]);
+        publisher->startPublishing(mPtPhysPrimDen[i]);
+        publisher->startPublishing(mFractionITSTPCmatchPhysPrim[i]);
+        publisher->startPublishing(mEtaPhysPrimNum[i]);
+        publisher->startPublishing(mEtaPhysPrimDen[i]);
+        publisher->startPublishing(mFractionITSTPCmatchEtaPhysPrim[i]);
+        publisher->startPublishing(m1OverPtPhysPrimNum[i]);
+        publisher->startPublishing(m1OverPtPhysPrimDen[i]);
+        publisher->startPublishing(mFractionITSTPCmatchPhysPrim1OverPt[i]);
+      }
     }
     publisher->startPublishing(mChi2Matching);
     publisher->startPublishing(mChi2Refit);
@@ -225,8 +232,6 @@ class MatchITSTPCQC
   void deleteHistograms();
   void setBz(float bz) { mBz = bz; }
 
-  // track selection
-  bool selectTrack(o2::tpc::TrackTPC const& track); // still present but not used
   // ITS track
   void setMinPtITSCut(float v) { mPtITSCut = v; };
   void setEtaITSCut(float v) { mEtaITSCut = v; }; // TODO: define 2 different values for min and max (**)

--- a/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx
+++ b/Detectors/GlobalTracking/src/MatchITSTPCQC.cxx
@@ -130,22 +130,16 @@ void MatchITSTPCQC::reset()
     mPtDen[i]->Reset();
     mPtNum_noEta0[i]->Reset();
     mPtDen_noEta0[i]->Reset();
-    mPtPhysPrimNum[i]->Reset();
-    mPtPhysPrimDen[i]->Reset();
 
     // Phi
     mPhiNum[i]->Reset();
     mPhiDen[i]->Reset();
-    mPhiPhysPrimNum[i]->Reset();
-    mPhiPhysPrimDen[i]->Reset();
     mPhiVsPtNum[i]->Reset();
     mPhiVsPtDen[i]->Reset();
 
     // Eta
     mEtaNum[i]->Reset();
     mEtaDen[i]->Reset();
-    mEtaPhysPrimNum[i]->Reset();
-    mEtaPhysPrimDen[i]->Reset();
     mEtaVsPtNum[i]->Reset();
     mEtaVsPtDen[i]->Reset();
 
@@ -156,6 +150,10 @@ void MatchITSTPCQC::reset()
     // Chi2
     mChi2VsPtNum[i]->Reset();
     mChi2VsPtDen[i]->Reset();
+
+    // 1/Pt
+    m1OverPtNum[i]->Reset();
+    m1OverPtDen[i]->Reset();
 
     if (mUseTrkPID) { // Vs Tracking PID hypothesis
       for (int j = 0; j < o2::track::PID::NIDs; ++j) {
@@ -170,12 +168,22 @@ void MatchITSTPCQC::reset()
         mEtaDenVsTrkPID[i][j]->Reset();
       }
     }
-    // 1/Pt
-    m1OverPtNum[i]->Reset();
-    m1OverPtDen[i]->Reset();
-    m1OverPtPhysPrimNum[i]->Reset();
-    m1OverPtPhysPrimDen[i]->Reset();
+
+    if (mUseMC) {
+      mPtPhysPrimNum[i]->Reset();
+      mPtPhysPrimDen[i]->Reset();
+
+      mPhiPhysPrimNum[i]->Reset();
+      mPhiPhysPrimDen[i]->Reset();
+
+      mEtaPhysPrimNum[i]->Reset();
+      mEtaPhysPrimDen[i]->Reset();
+
+      m1OverPtPhysPrimNum[i]->Reset();
+      m1OverPtPhysPrimDen[i]->Reset();
+    }
   }
+
   // Residuals
   mResidualPt->Reset();
   mResidualPhi->Reset();
@@ -192,81 +200,10 @@ void MatchITSTPCQC::reset()
 //__________________________________________________________
 bool MatchITSTPCQC::init()
 {
-
+  LOGP(debug, "Creating Variable Binning");
   std::array<std::string, 2> title{"TPC", "ITS"};
   std::array<std::string, 2> etaSel{"", ", |eta| < 0.9"};
   std::array<int, 2> maxNCls{156, 7};
-  for (int i = 0; i < matchType::SIZE; ++i) {
-    // Pt
-    mPtNum[i] = new TH1D(Form("mPtNum_%s", title[i].c_str()), Form("Pt distribution of ITSTPC matched tracks, wrt %s tracks %s; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f);
-    mPtDen[i] = new TH1D(Form("mPtDen_%s", title[i].c_str()), Form("Pt distribution of %s tracks %s; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f);
-    mFractionITSTPCmatch[i] = new TEfficiency(Form("mFractionITSTPCmatch_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks wrt %s tracks vs Pt %s; Pt [GeV/c]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f);
-    mPtNum_noEta0[i] = new TH1D(Form("mPtNum_noEta0_%s", title[i].c_str()), Form("Pt distribution of ITSTPC matched tracks without |eta| < 0.05, wrt %s tracks %s; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f);
-    mPtDen_noEta0[i] = new TH1D(Form("mPtDen_noEta0_%s", title[i].c_str()), Form("Pt distribution of %s tracks without |eta| < 0.05 %s; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f);
-    mFractionITSTPCmatch_noEta0[i] = new TEfficiency(Form("mFractionITSTPCmatch_noEta0_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks wrt %s tracks vs Pt without |eta| < 0.05 %s; Pt [GeV/c]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f);
-    // Phi
-    mPhiNum[i] = new TH1F(Form("mPhiNum_%s", title[i].c_str()), Form("Phi distribution of ITSTPC matched tracks, wrt %s tracks %s; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 2 * TMath::Pi());
-    mPhiDen[i] = new TH1F(Form("mPhiDen_%s", title[i].c_str()), Form("Phi distribution of %s tracks %s; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 2 * TMath::Pi());
-    mFractionITSTPCmatchPhi[i] = new TEfficiency(Form("mFractionITSTPCmatchPhi_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks vs Phi wrt %s tracks %s; Phi [rad]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 2 * TMath::Pi());
-    mPhiVsPtNum[i] = new TH2F(Form("mPhiVsPtNum_%s", title[i].c_str()), Form("Phi vs Pt distribution of ITSTPC matched tracks wrt %s %s; #it{p}_{T} [GeV#it{c}]; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f, 100, 0.f, 2 * TMath::Pi());
-    mPhiVsPtDen[i] = new TH2F(Form("mPhiVsPtDen_%s", title[i].c_str()), Form("Phi vs Pt distribution of %s tracks %s; #it{p}_{T} [GeV#it{c}]; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f, 100, 0.f, 2 * TMath::Pi());
-    mFractionITSTPCmatchPhiVsPt[i] = new TEfficiency(Form("mFractionITSTPCmatchPhiVsPt_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks wrt %s tracks %s, Phi vs Pt; #it{p}_{T} [GeV#it{c}]; Phi [rad]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f, 100, 0.f, 2 * TMath::Pi());
-    // Eta
-    mEtaNum[i] = new TH1F(Form("mEtaNum_%s", title[i].c_str()), Form("Eta distribution of ITSTPC matched tracks, wrt %s tracks; Eta; dNdEta", title[i].c_str()), 100, -2.f, 2.f);
-    mEtaDen[i] = new TH1F(Form("mEtaDen_%s", title[i].c_str()), Form("Eta distribution of %s tracks; Eta; dNdEta", title[i].c_str()), 100, -2.f, 2.f);
-    mFractionITSTPCmatchEta[i] = new TEfficiency(Form("mFractionITSTPCmatchEta_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks , wrt %s tracks, vs Eta; Eta; Eff", title[i].c_str()), 100, -2.f, 2.f);
-    mEtaVsPtNum[i] = new TH2F(Form("mEtaVsPtNum_%s", title[i].c_str()), Form("Eta vs Pt distribution of ITSTPC matched tracks, wrt %s tracks; #it{p}_{T} [GeV#it{c}]; Eta", title[i].c_str()), 100, 0.f, 20.f, 100, -2.f, 2.f);
-    mEtaVsPtDen[i] = new TH2F(Form("mEtaVsPtDen_%s", title[i].c_str()), Form("Eta vs Pt distribution of %s tracks; #it{p}_{T} [GeV#it{c}]; Eta", title[i].c_str()), 100, 0.f, 20.f, 100, -2.f, 2.f);
-    mFractionITSTPCmatchEtaVsPt[i] = new TEfficiency(Form("mFractionITSTPCmatchEtaVsPt_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks, wrt %s tracks, Eta vs Pt; #it{p}_{T} [GeV#it{c}]; Eta; Eff", title[i].c_str()), 100, 0.f, 20.f, 100, -2.f, 2.f);
-    // Clusters
-    mClsVsPtNum[i] = new TH2F(Form("mClsVsPtNum_%s", title[i].c_str()), Form("#Clusters vs Pt distribution of ITSTPC matched tracks, wrt %s tracks; #it{p}_{T} [GeV#it{c}]; #Clusters", title[i].c_str()), 100, 0.f, 20.f, maxNCls[i], 0, maxNCls[i]);
-    mClsVsPtDen[i] = new TH2F(Form("mClsVsPtDen_%s", title[i].c_str()), Form("#Clusters vs Pt distribution of %s tracks; #it{p}_{T} [GeV#it{c}]; #Clusters", title[i].c_str()), 100, 0.f, 20.f, maxNCls[i], 0, maxNCls[i]);
-    mFractionITSTPCmatchClsVsPt[i] = new TEfficiency(Form("mFractionITSTPCmatchClsVsPt_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks, wrt %s tracks, #Clusters vs Pt; #it{p}_{T} [GeV#it{c}]; #Clusters; Eff", title[i].c_str()), 100, 0.f, 20.f, maxNCls[i], 0, maxNCls[i]);
-    // Chi2
-    mChi2VsPtNum[i] = new TH2F(Form("mChi2VsPtNum_%s", title[i].c_str()), Form("Chi2 vs Pt distribution of ITSTPC matched tracks, wrt %s tracks; #it{p}_{T} [GeV#it{c}]; Chi2", title[i].c_str()), 100, 0.f, 20.f, 200, 0, 300);
-    mChi2VsPtDen[i] = new TH2F(Form("mChi2VsPtDen_%s", title[i].c_str()), Form("Chi2 vs Pt distribution of %s tracks; #it{p}_{T} [GeV#it{c}]; Chi2", title[i].c_str()), 100, 0.f, 20.f, 200, 0, 300);
-    mFractionITSTPCmatchChi2VsPt[i] = new TEfficiency(Form("mFractionITSTPCmatchChi2VsPt_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks, wrt %s tracks, Chi2 vs Pt; #it{p}_{T} [GeV#it{c}]; Chi2; Eff", title[i].c_str()), 100, 0.f, 20.f, 200, 0, 300);
-    if (mUseTrkPID) { // Vs Tracking PID hypothesis
-      for (int j = 0; j < o2::track::PID::NIDs; ++j) {
-        // Pt
-        mPtNumVsTrkPID[i][j] = new TH1D(Form("mPtNumVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Pt distribution of ITSTPC matched tracks, wrt %s tracks %s, TrkPID %i; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str(), j), 100, 0.f, 20.f);
-        mPtDenVsTrkPID[i][j] = new TH1D(Form("mPtDenVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Pt distribution of %s tracks %s, TrkPID %i; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str(), j), 100, 0.f, 20.f);
-        mFractionITSTPCmatchPtVsTrkPID[i][j] = new TEfficiency(Form("mFractionITSTPCmatchPtVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Fraction of ITSTPC matched tracks wrt %s tracks vs Pt %s, TrkPID %i; Pt [GeV/c]; Eff", title[i].c_str(), etaSel[i].c_str(), j), 100, 0.f, 20.f);
-        // Phi
-        mPhiNumVsTrkPID[i][j] = new TH1D(Form("mPhiNumVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Phi distribution of ITSTPC matched tracks, wrt %s tracks %s, TrkPID %i; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str(), j), 100, 0.f, 2 * TMath::Pi());
-        mPhiDenVsTrkPID[i][j] = new TH1D(Form("mPhiDenVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Phi distribution of %s tracks %s, TrkPID %i; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str(), j), 100, 0.f, 2 * TMath::Pi());
-        mFractionITSTPCmatchPhiVsTrkPID[i][j] = new TEfficiency(Form("mFractionITSTPCmatchPhiVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Fraction of ITSTPC matched tracks wrt %s tracks vs Phi %s, TrkPID %i; Phi [rad]; Eff", title[i].c_str(), etaSel[i].c_str(), j), 100, 0.f, 2 * TMath::Pi());
-        // Eta
-        mEtaNumVsTrkPID[i][j] = new TH1D(Form("mEtaNumVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Eta distribution of ITSTPC matched tracks, wrt %s tracks %s, TrkPID %i; Eta; dNdEta", title[i].c_str(), etaSel[i].c_str(), j), 100, -2.f, 2.f);
-        mEtaDenVsTrkPID[i][j] = new TH1D(Form("mEtaDenVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Eta distribution of %s tracks %s, TrkPID %i; Eta; dNdEta", title[i].c_str(), etaSel[i].c_str(), j), 100, -2.f, 2.f);
-        mFractionITSTPCmatchEtaVsTrkPID[i][j] = new TEfficiency(Form("mFractionITSTPCmatchEtaVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Fraction of ITSTPC matched tracks wrt %s tracks vs Eta %s, TrkPID %i; Eta; Eff", title[i].c_str(), etaSel[i].c_str(), j), 100, -2.f, 2.f);
-      }
-    }
-
-    // 1/pt
-    m1OverPtNum[i] = new TH1D(Form("m1OverPtNum_%s", title[i].c_str()), Form("1/Pt distribution of matched tracks, wrt %s tracks %s; 1/Pt [c/GeV]; dNdPt", title[i].c_str(), etaSel[i].c_str()), 100, -20.f, 20.f);
-    m1OverPtDen[i] = new TH1D(Form("m1OverPtDen_%s", title[i].c_str()), Form("1/Pt distribution of %s tracks %s; 1/Pt [c/GeV]; dNdPt", title[i].c_str(), etaSel[i].c_str()), 100, -20.f, 20.f);
-    mFractionITSTPCmatch1OverPt[i] = new TEfficiency(Form("mFractionITSTPCmatch1OverPt_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks vs 1/Pt, wrt %s tracks %s; 1/Pt [c/GeV]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, -20.f, 20.f);
-
-    // These will be empty in case of no MC info...
-    mPhiPhysPrimNum[i] = new TH1F(Form("mPhiPhysPrimNum_%s", title[i].c_str()), Form("Phi distribution of matched tracks (physical primary), wrt %s tracks %s; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 2 * TMath::Pi());
-    mPhiPhysPrimDen[i] = new TH1F(Form("mPhiPhysPrimDen_%s", title[i].c_str()), Form("Phi distribution of %s tracks (physical primary) %s; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 2 * TMath::Pi());
-    mFractionITSTPCmatchPhiPhysPrim[i] = new TEfficiency(Form("mFractionITSTPCmatchPhiPhysPrim_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks vs Phi (physical primary), wrt %s tracks %s; Phi [rad]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 2 * TMath::Pi());
-    mEtaPhysPrimNum[i] = new TH1F(Form("mEtaPhysPrimNum_%s", title[i].c_str()), Form("Eta distribution of matched tracks (physical primary), wrt %s tracks; Eta; dNdEta", title[i].c_str()), 100, -2.f, 2.f);
-    mEtaPhysPrimDen[i] = new TH1F(Form("mEtaPhysPrimDen_%s", title[i].c_str()), Form("Eta distribution of %s tracks (physical primary); Eta; dNdEta", title[i].c_str()), 100, -2.f, 2.f);
-    mFractionITSTPCmatchEtaPhysPrim[i] = new TEfficiency(Form("mFractionITSTPCmatchEtaPhysPrim_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks vs Eta (physical primary), wrt %s tracks; Eta; Eff", title[i].c_str()), 100, -2.f, 2.f);
-  }
-
-  mResidualPt = new TH2F("mResidualPt", "Residuals of ITS-TPC matching in #it{p}_{T}; #it{p}_{T}^{ITS-TPC} [GeV/c]; #it{p}_{T}^{ITS-TPC} - #it{p}_{T}^{TPC} [GeV/c]", 100, 0.f, 20.f, 100, -1.f, 1.f);
-  mResidualPhi = new TH2F("mResidualPhi", "Residuals of ITS-TPC matching in #it{#phi}; #it{#phi}^{ITS-TPC} [rad]; #it{#phi}^{ITS-TPC} - #it{#phi}^{TPC} [rad]", 100, 0.f, 2 * TMath::Pi(), 100, -1.f, 1.f);
-  mResidualEta = new TH2F("mResidualEta", "Residuals of ITS-TPC matching in #it{#eta}; #it{#eta}^{ITS-TPC}; #it{#eta}^{ITS-TPC} - #it{#eta}^{TPC}", 100, -2.f, 2.f, 100, -1.f, 1.f);
-  mChi2Matching = new TH1F("mChi2Matching", "Chi2 of matching; chi2", 200, 0, 300);
-  mChi2Refit = new TH1F("mChi2Refit", "Chi2 of refit; chi2", 200, 0, 300);
-  mDCAr = new TH1F("mDCAr", "DCA of TPC tracks; DCAr", 200, -100, 100);
-  mDCArVsPtNum = new TH2F("mDCArVsPtNum", "DCA of TPC tracks Vs Pt Num; #it{p}_{T} [GeV/c]; DCAr", 100, 0, 20., 200, -30, 30);
-  mDCArVsPtDen = new TH2F("mDCArVsPtDen", "DCA of TPC tracks Vs Pt Den; #it{p}_{T} [GeV/c]; DCAr", 100, 0, 20., 200, -30, 30);
-  mFractionITSTPCmatchDCArVsPt = new TEfficiency("mFractionITSTPCmatchDCArVsPt", "Fraction of ITSTPC matched tracks wrt TPC vs DCAr; #it{p}_{T} [GeV#it{c}]; DCAr; Eff", 100, 0, 20., 200, -30, 30);
-
   // log binning for pT
   const Int_t nbinsPt = 100;
   const Double_t xminPt = 0.01;
@@ -279,94 +216,151 @@ bool MatchITSTPCQC::init()
     Double_t xlogPt = xlogminPt + i * dlogxPt;
     xbinsPt[i] = TMath::Exp(TMath::Log(10) * xlogPt);
   }
-  mTimeResVsPt = new TH2F("mTimeResVsPt", "Time resolution vs Pt; Pt [GeV/c]; time res [us]", nbinsPt, xbinsPt, 100, 0.f, 2.f);
 
+  LOGP(debug, "Creating Histograms");
+  // Data and MC
   for (int i = 0; i < matchType::SIZE; ++i) {
-    mPtPhysPrimNum[i] = new TH1F(Form("mPtPhysPrimNum_%s", title[i].c_str()), Form("Pt distribution of matched tracks (physical primary), wrt %s tracks %s; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str()), nbinsPt, xbinsPt);
-    mPtPhysPrimDen[i] = new TH1F(Form("mPtPhysPrimDen_%s", title[i].c_str()), Form("Pt distribution of %s tracks (physical primary) %s; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str()), nbinsPt, xbinsPt);
-    mFractionITSTPCmatchPhysPrim[i] = new TEfficiency(Form("mFractionITSTPCmatchPhysPrim_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks vs Pt (physical primary), wrt %s tracks %s; Pt [GeV/c]; Eff", title[i].c_str(), etaSel[i].c_str()), nbinsPt, xbinsPt);
-    m1OverPtPhysPrimNum[i] = new TH1D(Form("m1OverPtPhysPrimNum_%s", title[i].c_str()), Form("1/Pt distribution of matched tracks (physical primary), wrt %s tracks %s; 1/Pt [c/GeV]; dNd1/Pt", title[i].c_str(), etaSel[i].c_str()), 100, -20.f, 20.f);
-    m1OverPtPhysPrimDen[i] = new TH1D(Form("m1OverPtPhysPrimDen_%s", title[i].c_str()), Form("1/PtPt distribution of %s tracks (physical primary) %s; 1/Pt [c/GeV]; dNd1/Pt", title[i].c_str(), etaSel[i].c_str()), 100, -20.f, 20.f);
-    mFractionITSTPCmatchPhysPrim1OverPt[i] = new TEfficiency(Form("mFractionITSTPCmatchPhysPrim1OverPt_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks vs 1/Pt (physical primary), wrt %s tracks %s; 1/Pt [c/GeV]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, -20.f, 20.f);
-
-    // some extra settings
     // Pt
+    mPtNum[i] = new TH1D(Form("mPtNum_%s", title[i].c_str()), Form("Pt distribution of ITSTPC matched tracks, wrt %s tracks %s; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f);
     mPtNum[i]->Sumw2();
+    mPtNum[i]->SetOption("logy");
+    mPtNum[i]->GetYaxis()->SetTitleOffset(1.4);
+    mPtDen[i] = new TH1D(Form("mPtDen_%s", title[i].c_str()), Form("Pt distribution of %s tracks %s; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f);
     mPtDen[i]->Sumw2();
+    mPtDen[i]->SetOption("logy");
+    mPtDen[i]->GetYaxis()->SetTitleOffset(1.4);
+    mFractionITSTPCmatch[i] = new TEfficiency(Form("mFractionITSTPCmatch_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks wrt %s tracks vs Pt %s; Pt [GeV/c]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f);
+    mPtNum_noEta0[i] = new TH1D(Form("mPtNum_noEta0_%s", title[i].c_str()), Form("Pt distribution of ITSTPC matched tracks without |eta| < 0.05, wrt %s tracks %s; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f);
     mPtNum_noEta0[i]->Sumw2();
+    mPtNum_noEta0[i]->SetOption("logy");
+    mPtNum_noEta0[i]->GetYaxis()->SetTitleOffset(1.4);
+    mPtDen_noEta0[i] = new TH1D(Form("mPtDen_noEta0_%s", title[i].c_str()), Form("Pt distribution of %s tracks without |eta| < 0.05 %s; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f);
     mPtDen_noEta0[i]->Sumw2();
-    mPtPhysPrimNum[i]->Sumw2();
-    mPtPhysPrimDen[i]->Sumw2();
+    mPtDen_noEta0[i]->SetOption("logy");
+    mPtDen_noEta0[i]->GetYaxis()->SetTitleOffset(1.4);
+    mFractionITSTPCmatch_noEta0[i] = new TEfficiency(Form("mFractionITSTPCmatch_noEta0_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks wrt %s tracks vs Pt without |eta| < 0.05 %s; Pt [GeV/c]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f);
 
     // Phi
+    mPhiNum[i] = new TH1F(Form("mPhiNum_%s", title[i].c_str()), Form("Phi distribution of ITSTPC matched tracks, wrt %s tracks %s; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 2 * TMath::Pi());
     mPhiNum[i]->Sumw2();
+    mPhiDen[i] = new TH1F(Form("mPhiDen_%s", title[i].c_str()), Form("Phi distribution of %s tracks %s; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 2 * TMath::Pi());
     mPhiDen[i]->Sumw2();
+    mFractionITSTPCmatchPhi[i] = new TEfficiency(Form("mFractionITSTPCmatchPhi_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks vs Phi wrt %s tracks %s; Phi [rad]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 2 * TMath::Pi());
+    mPhiVsPtNum[i] = new TH2F(Form("mPhiVsPtNum_%s", title[i].c_str()), Form("Phi vs Pt distribution of ITSTPC matched tracks wrt %s %s; #it{p}_{T} [GeV#it{c}]; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f, 100, 0.f, 2 * TMath::Pi());
     mPhiVsPtNum[i]->Sumw2();
+    mPhiVsPtDen[i] = new TH2F(Form("mPhiVsPtDen_%s", title[i].c_str()), Form("Phi vs Pt distribution of %s tracks %s; #it{p}_{T} [GeV#it{c}]; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f, 100, 0.f, 2 * TMath::Pi());
     mPhiVsPtDen[i]->Sumw2();
-    mPhiPhysPrimNum[i]->Sumw2();
-    mPhiPhysPrimDen[i]->Sumw2();
+    mFractionITSTPCmatchPhiVsPt[i] = new TEfficiency(Form("mFractionITSTPCmatchPhiVsPt_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks wrt %s tracks %s, Phi vs Pt; #it{p}_{T} [GeV#it{c}]; Phi [rad]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 20.f, 100, 0.f, 2 * TMath::Pi());
 
     // Eta
+    mEtaNum[i] = new TH1F(Form("mEtaNum_%s", title[i].c_str()), Form("Eta distribution of ITSTPC matched tracks, wrt %s tracks; Eta; dNdEta", title[i].c_str()), 100, -2.f, 2.f);
     mEtaNum[i]->Sumw2();
+    mEtaNum[i]->GetYaxis()->SetTitleOffset(1.4);
+    mEtaDen[i] = new TH1F(Form("mEtaDen_%s", title[i].c_str()), Form("Eta distribution of %s tracks; Eta; dNdEta", title[i].c_str()), 100, -2.f, 2.f);
     mEtaDen[i]->Sumw2();
-    mEtaPhysPrimNum[i]->Sumw2();
-    mEtaPhysPrimDen[i]->Sumw2();
+    mEtaDen[i]->GetYaxis()->SetTitleOffset(1.4);
+    mFractionITSTPCmatchEta[i] = new TEfficiency(Form("mFractionITSTPCmatchEta_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks , wrt %s tracks, vs Eta; Eta; Eff", title[i].c_str()), 100, -2.f, 2.f);
+    mEtaVsPtNum[i] = new TH2F(Form("mEtaVsPtNum_%s", title[i].c_str()), Form("Eta vs Pt distribution of ITSTPC matched tracks, wrt %s tracks; #it{p}_{T} [GeV#it{c}]; Eta", title[i].c_str()), 100, 0.f, 20.f, 100, -2.f, 2.f);
     mEtaVsPtNum[i]->Sumw2();
+    mEtaVsPtDen[i] = new TH2F(Form("mEtaVsPtDen_%s", title[i].c_str()), Form("Eta vs Pt distribution of %s tracks; #it{p}_{T} [GeV#it{c}]; Eta", title[i].c_str()), 100, 0.f, 20.f, 100, -2.f, 2.f);
     mEtaVsPtDen[i]->Sumw2();
+    mFractionITSTPCmatchEtaVsPt[i] = new TEfficiency(Form("mFractionITSTPCmatchEtaVsPt_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks, wrt %s tracks, Eta vs Pt; #it{p}_{T} [GeV#it{c}]; Eta; Eff", title[i].c_str()), 100, 0.f, 20.f, 100, -2.f, 2.f);
 
     // Clusters
+    mClsVsPtNum[i] = new TH2F(Form("mClsVsPtNum_%s", title[i].c_str()), Form("#Clusters vs Pt distribution of ITSTPC matched tracks, wrt %s tracks; #it{p}_{T} [GeV#it{c}]; #Clusters", title[i].c_str()), 100, 0.f, 20.f, maxNCls[i], 0, maxNCls[i]);
     mClsVsPtNum[i]->Sumw2();
+    mClsVsPtDen[i] = new TH2F(Form("mClsVsPtDen_%s", title[i].c_str()), Form("#Clusters vs Pt distribution of %s tracks; #it{p}_{T} [GeV#it{c}]; #Clusters", title[i].c_str()), 100, 0.f, 20.f, maxNCls[i], 0, maxNCls[i]);
     mClsVsPtDen[i]->Sumw2();
+    mFractionITSTPCmatchClsVsPt[i] = new TEfficiency(Form("mFractionITSTPCmatchClsVsPt_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks, wrt %s tracks, #Clusters vs Pt; #it{p}_{T} [GeV#it{c}]; #Clusters; Eff", title[i].c_str()), 100, 0.f, 20.f, maxNCls[i], 0, maxNCls[i]);
 
     // Chi2
+    mChi2VsPtNum[i] = new TH2F(Form("mChi2VsPtNum_%s", title[i].c_str()), Form("Chi2 vs Pt distribution of ITSTPC matched tracks, wrt %s tracks; #it{p}_{T} [GeV#it{c}]; Chi2", title[i].c_str()), 100, 0.f, 20.f, 200, 0, 300);
     mChi2VsPtNum[i]->Sumw2();
+    mChi2VsPtDen[i] = new TH2F(Form("mChi2VsPtDen_%s", title[i].c_str()), Form("Chi2 vs Pt distribution of %s tracks; #it{p}_{T} [GeV#it{c}]; Chi2", title[i].c_str()), 100, 0.f, 20.f, 200, 0, 300);
     mChi2VsPtDen[i]->Sumw2();
+    mFractionITSTPCmatchChi2VsPt[i] = new TEfficiency(Form("mFractionITSTPCmatchChi2VsPt_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks, wrt %s tracks, Chi2 vs Pt; #it{p}_{T} [GeV#it{c}]; Chi2; Eff", title[i].c_str()), 100, 0.f, 20.f, 200, 0, 300);
+
+    // 1/pt
+    m1OverPtNum[i] = new TH1D(Form("m1OverPtNum_%s", title[i].c_str()), Form("1/Pt distribution of matched tracks, wrt %s tracks %s; 1/Pt [c/GeV]; dNdPt", title[i].c_str(), etaSel[i].c_str()), 100, -20.f, 20.f);
+    m1OverPtNum[i]->Sumw2();
+    m1OverPtDen[i] = new TH1D(Form("m1OverPtDen_%s", title[i].c_str()), Form("1/Pt distribution of %s tracks %s; 1/Pt [c/GeV]; dNdPt", title[i].c_str(), etaSel[i].c_str()), 100, -20.f, 20.f);
+    m1OverPtDen[i]->Sumw2();
+    mFractionITSTPCmatch1OverPt[i] = new TEfficiency(Form("mFractionITSTPCmatch1OverPt_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks vs 1/Pt, wrt %s tracks %s; 1/Pt [c/GeV]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, -20.f, 20.f);
 
     if (mUseTrkPID) { // Vs Tracking PID hypothesis
       for (int j = 0; j < o2::track::PID::NIDs; ++j) {
         // Pt
+        mPtNumVsTrkPID[i][j] = new TH1D(Form("mPtNumVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Pt distribution of ITSTPC matched tracks, wrt %s tracks %s, TrkPID %i; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str(), j), 100, 0.f, 20.f);
         mPtNumVsTrkPID[i][j]->Sumw2();
+        mPtDenVsTrkPID[i][j] = new TH1D(Form("mPtDenVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Pt distribution of %s tracks %s, TrkPID %i; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str(), j), 100, 0.f, 20.f);
         mPtDenVsTrkPID[i][j]->Sumw2();
+        mFractionITSTPCmatchPtVsTrkPID[i][j] = new TEfficiency(Form("mFractionITSTPCmatchPtVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Fraction of ITSTPC matched tracks wrt %s tracks vs Pt %s, TrkPID %i; Pt [GeV/c]; Eff", title[i].c_str(), etaSel[i].c_str(), j), 100, 0.f, 20.f);
+
         // Phi
+        mPhiNumVsTrkPID[i][j] = new TH1D(Form("mPhiNumVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Phi distribution of ITSTPC matched tracks, wrt %s tracks %s, TrkPID %i; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str(), j), 100, 0.f, 2 * TMath::Pi());
         mPhiNumVsTrkPID[i][j]->Sumw2();
+        mPhiDenVsTrkPID[i][j] = new TH1D(Form("mPhiDenVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Phi distribution of %s tracks %s, TrkPID %i; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str(), j), 100, 0.f, 2 * TMath::Pi());
         mPhiDenVsTrkPID[i][j]->Sumw2();
+        mFractionITSTPCmatchPhiVsTrkPID[i][j] = new TEfficiency(Form("mFractionITSTPCmatchPhiVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Fraction of ITSTPC matched tracks wrt %s tracks vs Phi %s, TrkPID %i; Phi [rad]; Eff", title[i].c_str(), etaSel[i].c_str(), j), 100, 0.f, 2 * TMath::Pi());
+
         // Eta
+        mEtaNumVsTrkPID[i][j] = new TH1D(Form("mEtaNumVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Eta distribution of ITSTPC matched tracks, wrt %s tracks %s, TrkPID %i; Eta; dNdEta", title[i].c_str(), etaSel[i].c_str(), j), 100, -2.f, 2.f);
         mEtaNumVsTrkPID[i][j]->Sumw2();
+        mEtaDenVsTrkPID[i][j] = new TH1D(Form("mEtaDenVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Eta distribution of %s tracks %s, TrkPID %i; Eta; dNdEta", title[i].c_str(), etaSel[i].c_str(), j), 100, -2.f, 2.f);
         mEtaDenVsTrkPID[i][j]->Sumw2();
+        mFractionITSTPCmatchEtaVsTrkPID[i][j] = new TEfficiency(Form("mFractionITSTPCmatchEtaVsTrkPID_%s_PID%i", title[i].c_str(), j), Form("Fraction of ITSTPC matched tracks wrt %s tracks vs Eta %s, TrkPID %i; Eta; Eff", title[i].c_str(), etaSel[i].c_str(), j), 100, -2.f, 2.f);
       }
     }
-
-    m1OverPtNum[i]->Sumw2();
-    m1OverPtDen[i]->Sumw2();
-    m1OverPtPhysPrimNum[i]->Sumw2();
-    m1OverPtPhysPrimDen[i]->Sumw2();
-
-    mPtNum_noEta0[i]->SetOption("logy");
-    mPtDen_noEta0[i]->SetOption("logy");
-    mPtNum[i]->SetOption("logy");
-    mPtDen[i]->SetOption("logy");
-
-    mPtNum[i]->GetYaxis()->SetTitleOffset(1.4);
-    mPtDen[i]->GetYaxis()->SetTitleOffset(1.4);
-    mPtNum_noEta0[i]->GetYaxis()->SetTitleOffset(1.4);
-    mPtDen_noEta0[i]->GetYaxis()->SetTitleOffset(1.4);
-    mEtaNum[i]->GetYaxis()->SetTitleOffset(1.4);
-    mEtaDen[i]->GetYaxis()->SetTitleOffset(1.4);
   }
 
+  mResidualPt = new TH2F("mResidualPt", "Residuals of ITS-TPC matching in #it{p}_{T}; #it{p}_{T}^{ITS-TPC} [GeV/c]; #it{p}_{T}^{ITS-TPC} - #it{p}_{T}^{TPC} [GeV/c]", 100, 0.f, 20.f, 100, -1.f, 1.f);
+  mResidualPhi = new TH2F("mResidualPhi", "Residuals of ITS-TPC matching in #it{#phi}; #it{#phi}^{ITS-TPC} [rad]; #it{#phi}^{ITS-TPC} - #it{#phi}^{TPC} [rad]", 100, 0.f, 2 * TMath::Pi(), 100, -1.f, 1.f);
+  mResidualEta = new TH2F("mResidualEta", "Residuals of ITS-TPC matching in #it{#eta}; #it{#eta}^{ITS-TPC}; #it{#eta}^{ITS-TPC} - #it{#eta}^{TPC}", 100, -2.f, 2.f, 100, -1.f, 1.f);
+  mChi2Matching = new TH1F("mChi2Matching", "Chi2 of matching; chi2", 200, 0, 300);
   mChi2Matching->SetOption("logy");
-  mChi2Refit->SetOption("logy");
-  mTimeResVsPt->SetOption("colz logz logy logx");
-
   mChi2Matching->GetYaxis()->SetTitleOffset(1.4);
+  mChi2Refit = new TH1F("mChi2Refit", "Chi2 of refit; chi2", 200, 0, 300);
+  mChi2Refit->SetOption("logy");
   mChi2Refit->GetYaxis()->SetTitleOffset(1.4);
-  mTimeResVsPt->GetYaxis()->SetTitleOffset(1.4);
-
+  mDCAr = new TH1F("mDCAr", "DCA of TPC tracks; DCAr", 200, -100, 100);
+  mDCArVsPtNum = new TH2F("mDCArVsPtNum", "DCA of TPC tracks Vs Pt Num; #it{p}_{T} [GeV/c]; DCAr", 100, 0, 20., 200, -30, 30);
   mDCArVsPtNum->Sumw2();
+  mDCArVsPtDen = new TH2F("mDCArVsPtDen", "DCA of TPC tracks Vs Pt Den; #it{p}_{T} [GeV/c]; DCAr", 100, 0, 20., 200, -30, 30);
   mDCArVsPtDen->Sumw2();
+  mFractionITSTPCmatchDCArVsPt = new TEfficiency("mFractionITSTPCmatchDCArVsPt", "Fraction of ITSTPC matched tracks wrt TPC vs DCAr; #it{p}_{T} [GeV#it{c}]; DCAr; Eff", 100, 0, 20., 200, -30, 30);
+
+  mTimeResVsPt = new TH2F("mTimeResVsPt", "Time resolution vs Pt; Pt [GeV/c]; time res [us]", nbinsPt, xbinsPt, 100, 0.f, 2.f);
+  mTimeResVsPt->SetOption("colz logz logy logx");
+  mTimeResVsPt->GetYaxis()->SetTitleOffset(1.4);
 
   if (mUseMC) {
     mcReader.initFromDigitContext("collisioncontext.root");
+
+    for (int i = 0; i < matchType::SIZE; ++i) {
+      mPtPhysPrimNum[i] = new TH1F(Form("mPtPhysPrimNum_%s", title[i].c_str()), Form("Pt distribution of matched tracks (physical primary), wrt %s tracks %s; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str()), nbinsPt, xbinsPt);
+      mPtPhysPrimNum[i]->Sumw2();
+      mPtPhysPrimDen[i] = new TH1F(Form("mPtPhysPrimDen_%s", title[i].c_str()), Form("Pt distribution of %s tracks (physical primary) %s; Pt [GeV/c]; dNdPt", title[i].c_str(), etaSel[i].c_str()), nbinsPt, xbinsPt);
+      mPtPhysPrimDen[i]->Sumw2();
+      mFractionITSTPCmatchPhiPhysPrim[i] = new TEfficiency(Form("mFractionITSTPCmatchPhiPhysPrim_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks vs Phi (physical primary), wrt %s tracks %s; Phi [rad]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 2 * TMath::Pi());
+
+      mEtaPhysPrimNum[i] = new TH1F(Form("mEtaPhysPrimNum_%s", title[i].c_str()), Form("Eta distribution of matched tracks (physical primary), wrt %s tracks; Eta; dNdEta", title[i].c_str()), 100, -2.f, 2.f);
+      mEtaPhysPrimNum[i]->Sumw2();
+      mEtaPhysPrimDen[i] = new TH1F(Form("mEtaPhysPrimDen_%s", title[i].c_str()), Form("Eta distribution of %s tracks (physical primary); Eta; dNdEta", title[i].c_str()), 100, -2.f, 2.f);
+      mEtaPhysPrimDen[i]->Sumw2();
+      mFractionITSTPCmatchEtaPhysPrim[i] = new TEfficiency(Form("mFractionITSTPCmatchEtaPhysPrim_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks vs Eta (physical primary), wrt %s tracks; Eta; Eff", title[i].c_str()), 100, -2.f, 2.f);
+
+      mPhiPhysPrimNum[i] = new TH1F(Form("mPhiPhysPrimNum_%s", title[i].c_str()), Form("Phi distribution of matched tracks (physical primary), wrt %s tracks %s; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 2 * TMath::Pi());
+      mPhiPhysPrimNum[i]->Sumw2();
+      mPhiPhysPrimDen[i] = new TH1F(Form("mPhiPhysPrimDen_%s", title[i].c_str()), Form("Phi distribution of %s tracks (physical primary) %s; Phi [rad]; dNdPhi", title[i].c_str(), etaSel[i].c_str()), 100, 0.f, 2 * TMath::Pi());
+      mPhiPhysPrimDen[i]->Sumw2();
+      mFractionITSTPCmatchPhysPrim[i] = new TEfficiency(Form("mFractionITSTPCmatchPhysPrim_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks vs Pt (physical primary), wrt %s tracks %s; Pt [GeV/c]; Eff", title[i].c_str(), etaSel[i].c_str()), nbinsPt, xbinsPt);
+
+      m1OverPtPhysPrimNum[i] = new TH1D(Form("m1OverPtPhysPrimNum_%s", title[i].c_str()), Form("1/Pt distribution of matched tracks (physical primary), wrt %s tracks %s; 1/Pt [c/GeV]; dNd1/Pt", title[i].c_str(), etaSel[i].c_str()), 100, -20.f, 20.f);
+      m1OverPtPhysPrimNum[i]->Sumw2();
+      m1OverPtPhysPrimDen[i] = new TH1D(Form("m1OverPtPhysPrimDen_%s", title[i].c_str()), Form("1/PtPt distribution of %s tracks (physical primary) %s; 1/Pt [c/GeV]; dNd1/Pt", title[i].c_str(), etaSel[i].c_str()), 100, -20.f, 20.f);
+      m1OverPtPhysPrimDen[i]->Sumw2();
+      mFractionITSTPCmatchPhysPrim1OverPt[i] = new TEfficiency(Form("mFractionITSTPCmatchPhysPrim1OverPt_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks vs 1/Pt (physical primary), wrt %s tracks %s; 1/Pt [c/GeV]; Eff", title[i].c_str(), etaSel[i].c_str()), 100, -20.f, 20.f);
+    }
   }
 
   return true;
@@ -432,9 +426,6 @@ void MatchITSTPCQC::run(o2::framework::ProcessingContext& ctx)
 
   for (size_t itrk = 0; itrk < mTPCTracks.size(); ++itrk) {
     auto const& trkTpc = mTPCTracks[itrk];
-    // if (selectTrack(trkTpc)) {
-    //   isTPCTrackSelectedEntry[itrk] = true;
-    // }
     o2::dataformats::GlobalTrackID id(itrk, GID::TPC);
     if (cuts.isSelected(id, mRecoCont)) {
       // NB: same cuts for numerator and denominator tracks of ITS-TPC matching
@@ -876,31 +867,6 @@ void MatchITSTPCQC::run(o2::framework::ProcessingContext& ctx)
 }
 
 //__________________________________________________________
-
-bool MatchITSTPCQC::selectTrack(o2::tpc::TrackTPC const& track)
-{
-
-  if (track.getPt() < mPtCut) {
-    return false;
-  }
-  if (std::abs(track.getEta()) > mEtaCut) {
-    return false;
-  }
-  if (track.getNClusters() < mNTPCClustersCut) {
-    return false;
-  }
-
-  math_utils::Point3D<float> v{};
-  std::array<float, 2> dca{};
-  if (!(const_cast<o2::tpc::TrackTPC&>(track).propagateParamToDCA(v, mBz, &dca, mDCATPCCut)) || std::abs(dca[0]) > mDCATPCCutY) {
-    return false;
-  }
-
-  return true;
-}
-
-//__________________________________________________________
-
 void MatchITSTPCQC::finalize()
 {
 

